arangod/RocksDBEngine/RocksDBTransactionState.cpp

```diff
@@ -355,17 +356,14 @@ arangodb::Result RocksDBTransactionState::internalCommit() {
                _rocksTransaction->GetNumPuts() == 0 &&
                _rocksTransaction->GetNumDeletes() == 0);
     // this is most likely the fill index case
-    rocksdb::SequenceNumber seq = _rocksTransaction->GetSnapshot()->GetSequenceNumber();
+    #ifdef ARANGODB_ENABLE_MAINTAINER_MODE
     for (auto& trxColl : _collections) {
       TRI_IF_FAILURE("RocksDBCommitCounts") { continue; }
       auto* rcoll = static_cast<RocksDBTransactionCollection*>(trxColl);
-      rcoll->prepareCommit(id(), seq);
-      // We get here if we have filled indexes. So let us commit counts and
-      // any buffered index estimator updates
-      rcoll->commitCounts(id(), seq + 1);
```

Comment:
Are you sure we don't need to call `prepareCommit` and `commitCounts` any more?

Reply:
we never place blockers or anything else if there are no operations

Reply:
Ah, missed that context. Good.